### PR TITLE
[DoctrineBridge] Deprecate RegistryInterface

### PR DESCRIPTION
**Pull request by Koc**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yno
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | #32394
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: UPGRADE-5.0.md [hunk 1]
```diff
@@ -119,6 +119,7 @@ DoctrineBridge
    injected instead
  * Passing an `IdReader` to the `DoctrineChoiceLoader` when the query cannot be optimized with single id field will throw an exception, pass `null` instead
  * Not passing an `IdReader` to the `DoctrineChoiceLoader` when the query can be optimized with single id field will not apply any optimization
+ * The `RegistryInterface` has been removed.
```
Comment (nicolas-grekas) [inline UPGRADE-5.0.md]: don't miss UPGRADE-4.4 too

Comment (Koc) [reply UPGRADE-5.0.md]: done

File: UPGRADE-5.0.md [hunk 2]
```diff
@@ -119,6 +119,7 @@ DoctrineBridge
    injected instead
```
Comment (nicolas-grekas) [inline UPGRADE-5.0.md]: looks like this item should be on 4.4, and the message should be updated for 5.0, can you do it here too please?
please check the msg just below while doing so. thanks :)

**Review by nicolas-grekas — APPROVED** [review]

File: UPGRADE-5.0.md (hunk 2 above)
Comment (Koc) [reply UPGRADE-5.0.md]: I've added deprecation messages to 4.4 and updated them in 5.0. Please check them now.

File: UPGRADE-5.0.md [hunk 3]
```diff
@@ -115,10 +115,11 @@ DependencyInjection
 DoctrineBridge
 --------------
 
- * Deprecated injecting `ClassMetadataFactory` in `DoctrineExtractor`, an instance of `EntityManagerInterface` should be
+ * Removed possibility injecting `ClassMetadataFactory` in `DoctrineExtractor`, an instance of `EntityManagerInterface` should be
```
Comment (fabpot) [inline UPGRADE-5.0.md]: `Removed the possibility to inject`?

Comment (Koc) [reply UPGRADE-5.0.md]: done

**Review by Tobion — APPROVED** [review]

**Comment by Tobion** [comment]
Thank you @Koc.

**Pull request merged**